src/parse-result.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Parse result-specific refract elements.
+ *
+ * General structure:
+ *
+ * + ParseResult
+ *   + Annotation
+ */
+
+export function namespace(options) {
+  const minim = options.base;
+  const ArrayElement = minim.getElementClass('array');
+  const StringElement = minim.getElementClass('string');
+
+  class ParseResult extends ArrayElement {
+    constructor() {
+      super(...arguments);
+      this.element = 'parseResult';
+    }
+
+    get api() {
+      return this.children(item => item.classes.contains('api')).first();
+    }
+
+    get annotations() {
+      return this.children(item => item.element === 'annotation');
+    }
+
+    get warnings() {
+      return this.children(
+        item => item.element === 'annotation' &&
+        item.classes.contains('warning'));
+    }
+
+    get errors() {
+      return this.children(
+        item => item.element === 'annotation' &&
+        item.classes.contains('error'));
+    }
+  }
+
+  class Annotation extends StringElement {
+    constructor() {
+      super(...arguments);
+      this.element = 'annotation';
+    }
+
+    get code() {
+      return this.attributes.getValue('code');
+    }
+
+    set code(value) {
+      this.attributes.set('code', value);
+    }
+  }
+
+  class SourceMap extends minim.BaseElement {
```

Comment:
Given the current spec I think this is correct, so I'm going to leave it as-is, but maybe we need to revisit the spec with an RFC since really it is a specialized array element.